lib/fs/basicfs.go

```diff
@@ -29,6 +29,33 @@ type BasicFilesystem struct {
 }
 
 func newBasicFilesystem(root string) *BasicFilesystem {
+	root = cleanRoot(root)
```

Comment:
Does the symlink check potentially choke on a non-cleaned path? Otherwise this could just be moved after the symlink check to avoid calling it twice.

Reply:
I'm not familiar enough with the code base to answer this with confidence, but my suspicion is that it could for the same reasons that make the root cleaning necessary. Is the purpose of cleaning the root directory for Syncthing's internal assumptions for path handling or to ensure that calls to Go's filepath succeeds?

Reply:
The relevant commit for the code moved to cleanRoot is https://github.com/syncthing/syncthing/commit/3d8b4a42b72de26f6e15fdc2b2a7feeb116f9577

@AudriusButkevicius Could you add context for the path cleaning code and whether this 2 step check makes sense.

Reply:
In hindsight I am actually pretty sure you're right, at least for the tilde expansion. The other two `cleanRoot` changes (absolutification and the C:\ stuff) may or may not be necessary, but they are cheap, so unless Audrius can says with certainty that they are not necessary, lets just be save and use it twice as you do already.

Reply:
We should do it twice, in case eval symlinks returns a relative path.